WooCommerce/Classes/Authentication/AuthenticationManager.swift

```diff
@@ -183,6 +183,19 @@ extension AuthenticationManager: WordPressAuthenticatorDelegate {
     ///
     func createdWordPressComAccount(username: String, authToken: String) { }
 
+    func shouldDisplayError(_ error: Error) -> Bool {
+        return true
+    }
+
+    func handleError(_ error: Error, onCompletion: @escaping (WordPressAuthenticatorResult) -> Void) {
+        let viewModel = NotWPErrorViewModel(siteURL: "http://woocommerce.com")
```

Comment:
And here we instantiate custom UI and pass it back to WPAuthenticator. WPAuthenticator then will short-circuit its own error handling flow and just present the UI we provide

Reply:
It's probably fine if we just pass a view controller, instead of an instance of WordPressAuthenticatorResult to the completion closure. After all, host apps would only call this method *if* they want to insert custom UI to present the error